vault/token_store.go

```diff
@@ -1254,12 +1254,16 @@ func (ts *TokenStore) handleCreateCommon(
 	// the client has root or sudo privileges
 	case !isSudo:
 		// Sanitize passed-in and parent policies before comparison
-		sanitizedInputPolicies := policyutil.SanitizePolicies(data.Policies, true)
-		sanitizedParentPolicies := policyutil.SanitizePolicies(parent.Policies, true)
+		sanitizedInputPolicies := policyutil.SanitizePolicies(data.Policies, false)
+		sanitizedParentPolicies := policyutil.SanitizePolicies(parent.Policies, false)
 
 		if !strutil.StrListSubset(sanitizedParentPolicies, sanitizedInputPolicies) {
 			return logical.ErrorResponse("child policies must be subset of parent"), logical.ErrInvalidRequest
 		}
+
+		if !strutil.StrListContains(sanitizedParentPolicies, "default") {
+			data.NoDefaultPolicy = true
```

Comment:
What if a role is being used and the role doesn't explicitly list default? In fact, right now we always add default when a role is being used and either allowed policies or disallowed are set. We should probably take the no default flag into account in those cases as well.